src/articles/comments/comments.service.ts

```diff
@@ -0,0 +1,121 @@
+import { AuthorizationError, BadRequestError } from '@errors';
+import type { ProfilesService } from '@profiles/profiles.service';
+import type { UsersRepository } from '@users/users.repository';
+import type { CommentsRepository } from './comments.repository';
+import type { CommentToCreate, ReturnedComment } from './comments.schema';
+
+export class CommentsService {
+  constructor(
+    private readonly commentsRepository: CommentsRepository,
+    private readonly profilesService: ProfilesService,
+    private readonly usersRepository: UsersRepository,
+  ) {}
+
+  async createComment(
+    articleSlug: string,
+    commentBody: { body: string },
+    userId: number,
+  ): Promise<ReturnedComment> {
+    const article = await this.commentsRepository.findBySlug(articleSlug);
+
+    if (!article) {
+      throw new BadRequestError(`Article with slug ${articleSlug} not found`);
+    }
+
+    const commentData: CommentToCreate = {
+      ...commentBody,
+      authorId: userId,
+      articleId: article.id,
+    };
+
+    const comment = await this.commentsRepository.create(commentData);
+    const authorUsername = await this.getAuthorUsername(comment.authorId);
+    const authorProfile = await this.profilesService.findByUsername(
+      userId,
+      authorUsername,
+    );
```

Comment:
I think this should be done in one step, instead of querying the database twice. The profile service should probably have a findById method 🤔